daemon/api_general.go

```diff
@@ -130,13 +130,14 @@ func sysInfo(c *Command, r *http.Request, user *auth.UserState) Response {
 	}
 
 	m := map[string]interface{}{
-		"series":         release.Series,
-		"version":        c.d.Version,
-		"build-id":       buildID,
-		"os-release":     release.ReleaseInfo,
-		"on-classic":     release.OnClassic,
-		"managed":        len(users) > 0,
-		"kernel-version": osutil.KernelVersion(),
+		"series":          release.Series,
+		"version":         c.d.Version,
+		"build-id":        buildID,
+		"os-release":      release.ReleaseInfo,
+		"on-classic":      release.OnClassic,
+		"on-core-desktop": release.OnCoreDesktop,
```

Comment:
it feels like ultimately this belong more under os-release than here, we should probably not expose it here until we are sure of the way forward

Reply:
Regardless of where the information comes from having it explicitly stated here is useful so other code e.g. snapd-desktop-integration can match the behaviour of snapd.

Reply:
At present snapd-desktop-integration has no access to the `/v2/system-info` API endpoint, so this can't help there. And if there was an area where we wanted snapd-desktop-integration to change its behaviour to match snapd, it might be better to have snapd express that policy directly.

It looks like the only place `on-classic` is used at the moment is in formatting the `snap version` output (deciding whether to show the distro ID and version ID or not). If we wanted to add more information to that command, the question is whether we'd want to show `on-core-desktop`, or just the release variant.

Reply:
as I said my preference is to leave this out, at least until we understand more the behavior differences in snapd, but also as @jhenstridge mentioned this is not really available to snap-desktop-integration. It will be anyway reflected in the variant-id under os-release